controllers/controllers/workloads/shared.go

```diff
@@ -166,6 +171,35 @@ func propagateRoleBindings(ctx context.Context, kClient client.Client, log logr.
 		log.Info(fmt.Sprintf("Role Binding %s/%s %s", newRoleBinding.Namespace, newRoleBinding.Name, result))
 	}
 
+	propagatedRoleBindings := new(rbacv1.RoleBindingList)
+	labelSelector, err := labels.ValidatedSelectorFromSet(map[string]string{
+		korifiv1alpha1.PropagatedFromLabel: orgOrSpace.GetNamespace(),
+	})
+	if err != nil {
+		return err
+	}
+
+	err = kClient.List(ctx, propagatedRoleBindings, &client.ListOptions{Namespace: orgOrSpace.GetName(), LabelSelector: labelSelector})
+	if err != nil {
+		log.Error(err, fmt.Sprintf("Error listing role-bindings from namespace %s", orgOrSpace.GetName()))
+		return err
+	}
+
+	for _, propagatedBinding := range propagatedRoleBindings.Items {
+		toDelete := true
+		for _, parentBinding := range roleBindings.Items {
```

Comment:
It's probably better to have a map pre-created here so we can do this check with O(1) complexity

Reply:
Done